examples/habana/streaming_llm/run_streaming_llm.py

```diff
@@ -0,0 +1,167 @@
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# coding=utf-8
+# Copyright 2021 The EleutherAI and HuggingFace Teams. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import sys
+import argparse
+from typing import Any, Dict, List
+
+import torch
+from datasets import load_dataset
+from transformers import TextStreamer
+
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from run_generation import setup_parser
+from utils import print_memory_stats, initialize_model
+
+
+def create_prompts(samples: Dict[str, List[Any]]) -> Dict[str, Any]:
+    return {"prompt": [prompt for prompts in samples["prompt"] for prompt in prompts]}
+
+
+@torch.no_grad()
+def greedy_generate(model, tokenizer, dataset, args, generation_config, max_new_tokens=512, n_round=-1):
+    streamer = TextStreamer(tokenizer, skip_special_tokens=True)
+    new_line_tokens = tokenizer("\n\n", return_tensors="pt", add_special_tokens=False).input_ids
+    num_token = 0
+    count_round = 0
+
+    generation_config.max_new_tokens = max_new_tokens
+    generation_config.do_sample = False
+    generation_config.top_p = None
+    generation_config.use_cache = True
+    generation_config.attn_softmax_bf16 = True
+    generation_config.reuse_cache = True
+    generation_config.ignore_eos=False
+    generation_config.bucket_size = -1
```

Comment:
source?

Reply:
best speed here and must `use_cache` and `reuse_cache`. `do_sample=False` mean `greedy_search`, `top_p=None` will make `transformers` warning gone. `bucket_size` not used and don't supported. https://github.com/huggingface/optimum-habana/tree/main/examples/text-generation#running-with-fp8